tools/Dashboard/DevHome.Dashboard/Helpers/WidgetServiceHelper.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using Windows.Management.Deployment;
+
+namespace DevHome.Dashboard.Helpers;
+internal class WidgetServiceHelper
+{
+    private readonly Version minSupportedVersion400 = new (423, 3800);
+    private readonly Version minSupportedVersion500 = new (523, 3300);
+    private readonly Version version500 = new (500, 0);
+
+    private bool _validatedWebExpPack;
+
+    public WidgetServiceHelper()
+    {
+        _validatedWebExpPack = false;
+    }
+
+    public bool EnsureWebExperiencePack()
+    {
+        // If already validated there's a good version, don't check again.
+        if (_validatedWebExpPack)
+        {
+            return true;
+        }
+
+        // Ensure the application is installed, and the version is high enough.
+        const string packageName = "MicrosoftWindows.Client.WebExperience_cw5n1h2txyewy";
+
+        var packageManager = new PackageManager();

```

Comment:
Nit: This might be challenging to unit test in the future because `PackageManager` and its methods cannot be mocked here. Alternatively, we can add a testable and reusable method in the `PackageDeploymentService` service to check if a package version is installed. That would also require registering this helper class in the container.

Reply:
Resolving.